sources/Tools/Visualiser.py

```diff
@@ -126,25 +127,28 @@ def _process_bar_data(raw_data, head):
             "left": [(stat[index] == "Left").sum() for index in range(4)],
         }
 
-    def _histogram_visualizer(self, head):
-        logging.warning(
-            f"{inspect.currentframe().f_code.co_name}:Need to be refactor - No hard coded data please"
-        )
-        hands = self._process_bar_data(self.raw_data, head)
-        houses = set(self.raw_data.loc[:, "Hogwarts House"])
-        x = np.arange(len(houses))
-        width = 0.35
-        fig, ax = plt.subplots()
-        # print(hands)
-        rects1 = ax.bar(x - width / 2, hands["right"], width, label="Right")
-        rects2 = ax.bar(x + width / 2, hands["left"], width, label="Left")
-        ax.set_xticks(x)
-        ax.set_xticklabels(houses)
-        ax.legend()
-        self._auto_label(ax, rects1)
-        self._auto_label(ax, rects2)
-        fig.tight_layout()
-        return fig
+    @staticmethod
+    def _create_histogram(
+        feature, raw_data, filters, picker="Hogwarts House", xlabel="Marks", ylabel="Students"
+    ):
+        for elem in filters:
+            data = raw_data.loc[raw_data[picker] == elem, feature]
+            plt.hist(data, density=True, bins=30, alpha=0.5)
+            mn, mx = plt.xlim()
+            plt.xlim(mn, mx)
+            kde_xs = np.linspace(mn, mx, 301)
+            kde = st.gaussian_kde(data)
+            plt.plot(kde_xs, kde.pdf(kde_xs), label=elem)
+            plt.title(f"{elem} {feature}")
+        plt.xlabel(xlabel)
+        plt.ylabel(ylabel)
+        plt.yticks([])
+        plt.legend(filters, loc="upper left")
+        plt.show()

```

Comment:
don't show it please, save it or show it at the end of the program

Reply:
Sure, it was just the time to test the all plot program before rendering it in pdf files